File: LeetCode-Java/Missing Ranges.java
public class Solution {
    public List<String> findMissingRanges(int[] A, int lower, int upper) {
        List<String> result = new ArrayList<String>();
        int pre = lower;
        for(int i = 0;i <= A.length;i ++){
            int value = upper + 1;
            if(i < A.length)
                value = A[i];
            if(value - pre == 1){
                String tempResult = "" + pre;
                result.add(tempResult);
            }
            else if(value - pre > 1){
                String tempResult = pre + "->" + (value - 1);
                result.add(tempResult);
            }
            pre = value + 1;
            if(pre > upper)
                break;
        }
        return result;
    }
}

File: Leetcode-cpp/missing_ranges.cpp
class Solution {
public:
    vector<string> findMissingRanges(vector<int>& nums, int lower, int upper) {
        vector<string> result;
        nums.push_back(upper + 1);
        lower--;
        for (int i = 0; i < nums.size(); ++i) {
            if (nums[i] - lower == 2) {
                result.push_back(to_string(lower + 1));
            } else if (nums[i] - lower > 2) {
                result.push_back(to_string(lower + 1) + "->" + to_string(nums[i] - 1));
            }
            lower = max(lower, nums[i]);
        }
        nums.erase(nums.end() - 1);
        return result;
    }
};